app/controllers/works_controller.rb

```diff
@@ -2,8 +2,11 @@ class WorksController < ApplicationController
   # We should always be able to tell what category
   # of work we're dealing with
   before_action :category_from_work, except: [:root, :index, :new, :create]
+  before_action :crefator, only:[:edit, :update, :destroy]
+  skip_before_action :require_login, only: [:root]
 
```

Comment:
Looks like this is a typo - you've got `crefator` here, which is definitely wrong. As is, this makes it so when any user attempts to update, edit or delete a book they hit an error.

When I fixed the typo, things seemed to work as expected.